dali/operators/reader/loader/video/frames_decoder.cc

```diff
@@ -206,8 +206,9 @@ FramesDecoder::FramesDecoder(const std::string &filename)
 
 
 FramesDecoder::FramesDecoder(const char *memory_file, int memory_file_size, bool build_index,
-                             bool init_codecs, int num_frames)
+                             bool init_codecs, int num_frames, const std::string &source_info)
   : av_state_(std::make_unique<AvState>()),
+    filename_(source_info.size() ? std::optional<const std::string>(source_info) : std::nullopt),
```

Comment:
This whole `std::optional<const std::string>` affair seems unnecessarily complex. I don't think it's possible to have a meaningful empty filename - so we could just use empty string to denote the "memory file" (see frames_decoder.h:202, the only usage of filename_.value). This would simplify the code here, too.

Reply:
Done